location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,577 @@
+        start_time: The start time in seconds of the labelled video segment.
+        end_time: The end time in seconds of the labelled video segment.
+    """
+    tmp_path = join(dest_dir, 'tmp')
+    if isdir(tmp_path):
+        length = len(listdir(tmp_path))
+        m4a_path = join(tmp_path, video_id + '.m4a')
+        opus_path = join(tmp_path, video_id + '.opus')
+        ogg_path = join(tmp_path, video_id + '.ogg')
+        wav_path = join(dest_dir, 'yt_videos', 'sliced_' + video_id + '.wav')
+        if isfile(m4a_path):
+            video = AudioSegment.from_file(m4a_path)
+            temp_path = m4a_path
+        elif isfile(opus_path):
+            video = AudioSegment.from_file(opus_path)
+            temp_path = opus_path
+        elif isfile(ogg_path):
+            video = AudioSegment.from_file(ogg_path)
+            temp_path = ogg_path
+        else:
+            if length == 0:
+                shutil.rmtree(tmp_path)
+            return None
+        sliced = video[start_time * 1000: end_time * 1000]
+        sliced.export(wav_path, format='wav')
+        os.remove(temp_path)
+        if length <= 1:
+            shutil.rmtree(tmp_path)
+        logging.info('chopped_audio')
+
+
+def extract_feature(dest_dir, video_id, feature, scaled):
+    """Extracts a feature from a specific audio file given a video_id.
+
+   Given a specific of video_id, it extracts features using the Librosa libray
+   from the corresponding audio file and stores the features in a dictionary
+   with a video_id, dictionary key, value pair, with the dictionary value being
+   comprised of feature name, feature list key, value pairs. Using Librosa it
+   extracts one of the following features: chroma_stft, chroma_cqt, chroma_cens,
+   melspectogram, mfcc, rms, spectral_centroid, spectral_bandwidth,
+   spectral_contrast, spectral_flatness, spectral_rolloff, poly_features,
+   tonnetz, zero_crossing_rate
+
+   Args:
+       dest_dir: Path to the parent directory where the downloaded videos are
+           stored.
+       video_id: The video_id of specific audio file from which features are to
+           be extracted.
+       feature: A dictionary with video_id, dictionary pairs with the
+           dictionary values being feature name, feature list key, value pairs.
+       scaled: whether the extracted feature should be averaged
+
+   Returns:
+       A boolean of whether the extracting features was successful
+
+   Raises: ValueError: A feature not supported by Librosa has been inputted.
+   """
+    path = join(dest_dir, 'yt_videos', 'sliced_' + video_id + '.wav')
+    if not isfile(path):
+        return None
+    try:
+        # get audio (y) and sampling rate (sr)
+        y, sr = librosa.load(dest_dir + '/yt_videos/sliced_'
+                             + video_id + '.wav')
+    except ValueError as error:
+        logging.error(error)
+        return False
+    switcher = {
```

Comment:
Instead of `switcher`, consider `feature_name_dict`

Reply:
Done.